### DISCUSSION TITLE
Gitlab : GitLab GraphQL success?

### DISCUSSION THREAD

**Discussion by nitrams0** [grafana/grafana-infinity-datasource#110] [Specific APIs]
I'm using Infinity 0.7.0 GraohQL and trying to query GitLab's GraphQL. I successfully returned data from GitLab using the GraphQL datasource.

From the datasource UI:
   I've set the URL as https://gitlab.com/api/graphql
   I've added two headers:
      Content-Type: application/json
      Authorization: Bearer <GRAPHQL_TOKEN>

From within the target panel, I set:
   Type = GraphQL
   Source = URL
   Format = Table

Under URL Options:
Method = POST
Body = 
{
   currentUser {
      id
   }
}

Under Query Inspector:
When I oress the Refresh button, it just spins. I get nothing.

If I move all my URL info to the Panel itself, I get a 404 error and this under the error:
404 Not Found
Object
status:500
statusText:"Internal Server Error"
data:Object
error:"Internal Server Error"
response:"404 Not Found
"
message:"404 Not Found
"
config:Object
method:"POST"
url:"api/datasources/2/resources/proxy"
data:Object
retry:0
headers:Object
hideFromInspector:true
message:"404 Not Found
"
Does anyone have any tips or words of wisdom? Any info is greatly appreciated.

Thanks!

**Comment by yesoreyeram** [comment]
### In the config 

URL : `https://gitlab.com/api/graphql`
Enable authentication : Yes
Username : Leave blank
Password : `Your Gitlab personal access token`
> here in the config, you no need to set any headers. You gitlab PAT is your password and it doesn't have username.

### In the query

change method to `POST`, leave URL blank and enter your query. Hope this helps.

![image](https://user-images.githubusercontent.com/153843/127625759-a812de11-6d99-4bdc-8500-b8eebdcaf031.png)
![image](https://user-images.githubusercontent.com/153843/127625862-ee6dec4a-c5a6-4e08-af0c-1f56c5b0081a.png)

**Reply by nitrams0** [reply]
Well, I followed your instructions and still am getting no return from the Query Inspector
Plugin config:
![image](https://user-images.githubusercontent.com/3441533/127886112-e99b1162-06e4-4915-a7de-4ddda6ddcb86.png)

Query Config:
![image](https://user-images.githubusercontent.com/3441533/127886343-09c81a7b-637e-4b50-82a8-cc9ce2776dd6.png)

Query Inspector:
![image](https://user-images.githubusercontent.com/3441533/127886388-408feccc-26a5-47d7-b1df-f121202936d0.png)

**Reply by yesoreyeram** [reply]
Can you check `Data` tab instead `Query`? Also I suggest to change it to `Table` view instead graph/timeseries panel.

**Reply by yesoreyeram** [reply]
or browser network tab? `F12` or developer tools?
